atmcfg/atmcfg_test.go

```diff
@@ -62,6 +64,156 @@ func TestStartup(t *testing.T) {
 	})
 }
 
+func TestShutdownProcessesByClusterName(t *testing.T) {
+	t.Run("replica set", func(t *testing.T) {
+		config := automationConfigWithOneReplicaSet(clusterName, false)
+		err := ShutdownProcessesByClusterName(config, clusterName, []string{"host0:27017"})
+		if err != nil {
+			t.Error(err)
+		}
+		if !config.Processes[0].Disabled {
+			t.Errorf("ShutdownProcess\n got=%#v\nwant=%#v\n", config.Processes[0].Disabled, true)
+		}
+	})
+	t.Run("sharded cluster - one process", func(t *testing.T) {
+		config := automationConfigWithOneShardedCluster(clusterName, false)
+		err := ShutdownProcessesByClusterName(config, clusterName, []string{"host2:27018"})
+		if err != nil {
+			t.Error(err)
+		}
+
+		if config.Processes[0].Disabled {
+			t.Errorf("TestShutdownClusterAndProcesses\n got=%#v\nwant=%#v\n", config.Processes[0].Disabled, false)
+		}
+
+		if !config.Processes[1].Disabled {
+			t.Errorf("TestShutdownClusterAndProcesses\n got=%#v\nwant=%#v\n", config.Processes[1].Disabled, true)
+		}
+	})
+	t.Run("sharded cluster - two processes", func(t *testing.T) {
+		config := automationConfigWithOneShardedCluster(clusterName, false)
+		err := ShutdownProcessesByClusterName(config, clusterName, []string{"host2:27018", "host0:27017"})
+
+		if err != nil {
+			t.Error(err)
+		}
+
+		if !config.Processes[0].Disabled {
+			t.Errorf("TestShutdownClusterAndProcesses\n got=%#v\nwant=%#v\n", config.Processes[0].Disabled, true)
+		}
+
+		if !config.Processes[1].Disabled {
+			t.Errorf("TestShutdownClusterAndProcesses\n got=%#v\nwant=%#v\n", config.Processes[1].Disabled, true)
+		}
+	})
+	t.Run("shutdown entire sharded cluster", func(t *testing.T) {
+		config := automationConfigWithOneShardedCluster(clusterName, true)
+
+		err := ShutdownProcessesByClusterName(config, clusterName, nil)
+		if err != nil {
+			t.Error(err)
+		}
+
+		for i := range config.Processes {
+			if !config.Processes[i].Disabled {
+				t.Errorf("TestShutdownClusterAndProcesses\n got=%#v\nwant=%#v\n", config.Processes[i].Disabled, true)
+			}
+		}
+	})
+
+	t.Run("provide a process that does not exist", func(t *testing.T) {
+		config := automationConfigWithOneShardedCluster(clusterName, true)
+
+		err := ShutdownProcessesByClusterName(config, clusterName, []string{"hostTest:21021"})
+		expected := fmt.Sprintf(`process not found in "%v": %s`, clusterName, "[hostTest:21021]")
+
+		var e *ProcessNotFoundError
+
+		if !errors.As(err, &e) {
```

Comment:
[nit] use `Is` to check types, `As` is for casting
```suggestion
		if !errors.Is(err, &e) {
```